Patch pprint to make pytest diffs nicer for big objects

alexmojaki opened the pull request:
Closes https://github.com/samuelcolvin/dirty-equals/issues/73

Based on the branch in https://github.com/samuelcolvin/dirty-equals/pull/81

alexmojaki committed:
Patch pprint to make pytest diffs nicer for big objects

alexmojaki committed:
comments

samuelcolvin commented on tests/test_base.py:
```diff
@@ -1,12 +1,17 @@
 from dirty_equals.version import VERSION
 
 
+def check_repr_pformat(v, v_repr):
```
Better to keep these inline to improve the dev experience.

samuelcolvin reviewed (commented):
Otherwise lgtm

alexmojaki committed:
inline check_repr_pformat

evstratbg committed:
change license to exact (#85)

Hi! 

Thanks for the great tool.

In our system, we run checks on the licences in the libraries we use and
having `file` here adds some complexity in parsing.
Would it be possible to have a certain license here, as proposed?

---------

Co-authored-by: Samuel Colvin <s@muelcolvin.com>

MeggyCal committed:
fix DeprecationWarning with datetime.utcnow() (#86)

This is not the same as
https://github.com/samuelcolvin/dirty-equals/issues/71, but it
originates at the same point. When I updated pytz to 2023.3.post1, I
started getting:
```
E       DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
```
This fixes it.

---------

Co-authored-by: Samuel Colvin <s@muelcolvin.com>

alexmojaki committed:
Make datetime tests pass without requiring UTC system timezone  (#81)

...and a few other things that got in my way as an initial contributor.

Closes https://github.com/samuelcolvin/dirty-equals/issues/33

alexmojaki committed:
Merge branch 'main' of github.com:samuelcolvin/dirty-equals into pprint

samuelcolvin committed:
uprev dependencies and use pydantic v2 (#88)

fix #72

Ended up having to do a bunch of things at the same time to get tests to
pass:
* drop 3.7 (added 3.12 at the same time)
* uprev deps
* uprev pydantic requirement to >2
* switch `IsUrl` to assume pydantic 2

samuelcolvin committed:
support python 3.12 (#89)

samuelcolvin committed:
fix main docs build

samuelcolvin committed:
better dep constraints (#90)

samuelcolvin committed:
fix coverage after dropping 3.7 (#91)

samuelcolvin committed:
uprev

alexmojaki committed:
Merge branch 'main' of github.com:samuelcolvin/dirty-equals into pprint

alexmojaki committed:
fix merge
